Tests/MessagingInApp/MessagingInAppImplementationTest.swift

```diff
@@ -135,171 +165,38 @@ class MessagingInAppImplementationTest: IntegrationTest {
 
         await waitForExpectations(expectAsyncEventBusEvents)
 
-        XCTAssertEqual(inAppProviderMock.setRouteCallsCount, 1)
-        XCTAssertEqual(inAppProviderMock.setRouteReceivedArguments, given)
+        XCTAssertEqual(gistProviderMock.setCurrentRouteCallsCount, 1)
+        XCTAssertEqual(gistProviderMock.setCurrentRouteReceivedArguments, given)
     }
 
     // MARK: event listeners
 
-    func test_eventListeners_expectCallListenerWithData() async {
+    func test_eventListeners_expectNoCallListenerWithData() async {
         await waitForExpectations(initializeModule())
 
-        let givenGistMessage = Message.random
-        let expectedInAppMessage = InAppMessage(gistMessage: givenGistMessage)
-
         messagingInApp.setEventListener(eventListenerMock)
 
         XCTAssertFalse(eventListenerMock.messageShownCalled)
-        messagingInApp.messageShown(message: givenGistMessage)
-        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 1)
-        XCTAssertEqual(eventListenerMock.messageShownReceivedArguments, expectedInAppMessage)
 
         // message dismissed
         XCTAssertFalse(eventListenerMock.messageDismissedCalled)
-        messagingInApp.messageDismissed(message: givenGistMessage)
-        XCTAssertEqual(eventListenerMock.messageDismissedCallsCount, 1)
-        XCTAssertEqual(eventListenerMock.messageDismissedReceivedArguments, expectedInAppMessage)
 
         // error with message
         XCTAssertFalse(eventListenerMock.errorWithMessageCalled)
-        messagingInApp.messageError(message: givenGistMessage)
-        XCTAssertEqual(eventListenerMock.errorWithMessageCallsCount, 1)
-        XCTAssertEqual(eventListenerMock.errorWithMessageReceivedArguments, expectedInAppMessage)
 
         // message action taken
         XCTAssertFalse(eventListenerMock.messageActionTakenCalled)
-        let givenCurrentRoute = String.random
-        let givenAction = String.random
-        let givenName = String.random
-        messagingInApp.action(
-            message: givenGistMessage,
-            currentRoute: givenCurrentRoute,
-            action: givenAction,
-            name: givenName
-        )
-        XCTAssertEqual(eventListenerMock.messageActionTakenCallsCount, 1)
-        XCTAssertEqual(eventListenerMock.messageActionTakenReceivedArguments?.message, expectedInAppMessage)
-        XCTAssertEqual(eventListenerMock.messageActionTakenReceivedArguments?.actionValue, givenAction)
-        XCTAssertEqual(eventListenerMock.messageActionTakenReceivedArguments?.actionName, givenName)
-    }
-
-    func test_eventListeners_expectCallListenerForEachEvent() async {
```

Comment:
this is now the responsibility of `Gist` so these tests will be added to its test suite,